prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/ModelNexus.java

```diff
@@ -469,14 +499,21 @@ private ModelTransaction getOrCreateResource(String resource, Class<?> type, Mod
             if (transaction.getServiceState() != ModelTransactionState.NEW) {
                 transaction.setResourceState(ModelTransactionState.NEW);
             }
-            updateMetadata(service);
-            feature = EMFUtil.createEAttribute(service, resource, type, ef -> createEFeatureAnnotation(ef, timestamp));
-            service.getEStructuralFeatures().add(feature);
+            feature = createResource(service, resource, type, timestamp, null, accumulator);
         }
         transaction.addFeature(feature);
         return transaction;
     }
 
+    private EAttribute createResource(EClass service, String resource, Class<?> type, Instant timestamp,
+            Object defaultValue, NotificationAccumulator accumulator) {
```

Comment:
Do we need the accumulator here?

Reply:
We will do - the notifications are missing some events at the moment, including linking/unlinking and model change events